chartsxblock/static/js/src/chartsxblock-studio.js

```diff
@@ -37,36 +99,117 @@ function BadChartError(message) {
 }
 
 BadChartError.prototype = Object.create(Error.prototype);
-BadChartError.prototype.name = "BadChartError";
 BadChartError.prototype.constructor = BadChartError;
 
 function size(ar){
     var row_count = ar.length;
     var row_sizes = []
     for(var i=0;i<row_count;i++){
-        row_sizes.push(ar[i].length)
+        row_sizes.push(ar[i].length);
     }
-    if(Math.min.apply(null, row_sizes) != Math.max.apply(null, row_sizes)) throw new BadChartError("")
-    return [row_count, Math.min.apply(null, row_sizes)]
+    if(Math.min.apply(null, row_sizes) != Math.max.apply(null, row_sizes))
+        throw new BadChartError("All chart rows are not equal length.");
+    return [row_count, Math.min.apply(null, row_sizes)];
 }
-*/
 
 function ValidateData(inputString) {
-    //TODO
     try {
         parsedString = JSON.parse(inputString);
     } catch (e) {
         throw e;
     }
+    chartDimensions = size(parsedString);
+    if (chartDimensions[0] < 2) throw new BadChartError("The chart requires more than one row.")
     return inputString;
 }
 
-function ValidateType(inputString) {
-    //TODO
+function ValidateType(inputString, chartTypes) {
+    if(chartTypes.indexOf(inputString) == -1 ) throw new BadChartError("Using an unsupported chart type.");
     return inputString;
 }
 
 function ValidateName(inputString) {
-    //TODO
+    if(inputString.length < 1) throw new BadChartError("Chart name is empty.");
     return inputString;
 }
+
+function ChartTable(targetDiv, inputJSON){
+    this.data = JSON.parse(inputJSON);
+    this.rows = this.data.length;
+    this.columns = this.data[0].length;
+    this.div = targetDiv;
+
+    this.draw = function() {
+    // Draws the table in a div
+        for (i = 0; i < this.data.length; i++) {
+            var row = $("<tr>");
+            for (j = 0; j < this.data[i].length; j++) {
+                var field = $("<td>" + this.data[i][j] + "</td>");
+                row.append(field);
+            }
+            this.div.find("tbody").append(row);
+        }
+    }
+
+    this.update = function() {
+    // Updates the table from its div element
+        var dataArray = [];
+        var parent = this;
+        this.div.find("tbody").children("tr").each(function() {
+            var row = [];
+            $(this).children("td").each(function() {
+                row.push(parent.parse($(this).text()));
+            });
+            dataArray.push(row);
+        });
+        this.data = dataArray;
+    }
+
+    this.parse = function(val) {
```

Comment:
Pa kombinirao sam parseFloat i parseInt, kao što si sugerirao.

Ne mogu ostaviti bez provjere jer će onda stringovi biti NaN, te ih neću
moć predati u Charts, što je u nekim slučajevima korisno.

On 11 Aug 2016 3:40 pm, "Vedran Karačić" <notifications@github.com> wrote:

In chartsxblock/static/js/src/chartsxblock-studio.js
<https://github.com/TinMarkovic/ChartsXBlock/pull/1#discussion_r74423226>:

> +
> +    this.update = function() {
> +    // Updates the table from its div element
> +        var dataArray = [];
> +        var parent = this;
> +        this.div.find("tbody").children("tr").each(function() {
> +            var row = [];
> +            $(this).children("td").each(function() {
> +                row.push(parent.parse($(this).text()));
> +            });
> +            dataArray.push(row);
> +        });
> +        this.data = dataArray;
> +    }
> +
> +    this.parse = function(val) {

Nisi odgovorio na ovo.

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/TinMarkovic/ChartsXBlock/pull/1/files/32d5d49382d03f26b9983838493f3d7d0fb12ab7#r74423226>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AL7p0OvaNJrRTLUouNYiuNHe15a44JsMks5qeyZBgaJpZM4Jbm8G>
.